pinot-minion/src/main/java/com/linkedin/pinot/minion/executor/BaseTaskExecutor.java

```diff
@@ -32,4 +40,22 @@ public void setMinionContext(@Nonnull MinionContext minionContext) {
   public void cancel() {
     _cancelled = true;
   }
+
+  @Override
+  public int uploadSegment(String uri, final String fileName, final InputStream inputStream, final long lengthInBytes,
+      FileUploadUtils.SendFileMethod httpMethod, String originalSegmentCrc, String jobType) {
+    List<Header> headers = new ArrayList<>();
+    Header ifMatchHeader = new Header(HttpHeaders.IF_MATCH, originalSegmentCrc);
+    HttpParams httpParams = DefaultHttpParams.getDefaultParams();
+    String userAgentParameter = String.valueOf(httpParams.getParameter("http.useragent"));
```

Comment:
please use a static final string  for "http.useragent", better yet, try to find the definition for this string in the HttpParams (or another) object.